core/src/main/java/com/sequenceiq/cloudbreak/service/image/userdata/UserDataService.java

```diff
@@ -84,8 +89,9 @@ public void createUserData(Long stackId) throws CloudbreakImageNotFoundException
         try {
             PlatformParameters platformParameters = platformParametersFuture.get();
             CcmParameters ccmParameters = fetchCcmParameters(stack);
+            Optional<ProxyConfig> proxyConfig = proxyConfigDtoService.getByEnvironmentCrn(stack.getEnvironmentCrn());
             Map<InstanceGroupType, String> userData = userDataBuilder.buildUserData(Platform.platform(stack.getCloudPlatform()), cbSshKeyDer,
-                    sshUser, platformParameters, saltBootPassword, cbCert, ccmParameters);
+                    sshUser, platformParameters, saltBootPassword, cbCert, ccmParameters, proxyConfig.get());
```

Comment:
so I feel some inconsistency here
You have an optional and call get without checking if it's present and then do a null check in userdatabuilder. I suggest to pass the optional as param and replace the null check with isPresent

Reply:
Optional.orElse(null) will be used instead of Optional.get(). The Optional as method parameter is not recommended, so I wouldn't use it.